lib/susedistribution.pm

```diff
@@ -108,6 +108,18 @@ sub x11_start_program($$$) {
         record_info('workaround', 'desktop-runner does not show up on alt-f2, retrying up to three times (see bsc#978027)');
         send_key_until_needlematch 'desktop-runner', 'alt-f2', 3, 10;
     }
+    #check if desktop runner is focussed to type in
+    #currently only needle for gnome
+    if (match_has_tag 'desktop-runner-unfocused') {
+        if (check_var('DESKTOP', 'gnome')) {
+            record_soft_failure('boo#1046320');
+        }
+        else {
+            die("unexpected boo#1046320 in non-gnome");
```

Comment:
Why not to do this only if it's gnome and match_has_tag 'desktop-runner-unfocused', so test will fail with assert_screen and one can easily spot the issue?